src/java/org/apache/cassandra/net/OutboundTcpConnection.java

```
@@ -510,7 +510,6 @@ private boolean connect()
                 socket = null;
                 if (DatabaseDescriptor.isSslNteEnabled()) {
                     StorageService.instance.recordNonTransientError(StorageServiceMBean.NonTransientError.SSL_ERROR, ImmutableMap.of());
```

recordTransientError?

---

lol so I'm torn on this, just b/c an ssl error _could_ be transient, or non-transient. I'd argue it's more so the latter, as the only `transient` form of this is a certificate issued too early (which will keep occurring, until the issuer manually fixes this).

---

this is mostly up to us to define, cause we don't actually have transient anywhere else defined.  i think "could be resolved with a bounce" probably means it's transient, because the contrapositive "non-transient" explicitly means _requires manual action to resolve_.  agree that sometimes a bounce won't help though, but then the question is, is there an automated action we could take to resolve?

ideally we would have the node that is the _source_ be non transient and the rest of the nodes are transient but that might be hard

---

I'd push for keeping this non-transient then, as I for-see the majority of our SSL issues requiring manual action, as it indicates that certificates are wrong in some sort of sense.

 I.e., we're issuing certificates too early --> need to manually fix what's generating the certs.
Certs are missing hostname entries --> won't be fixed until a code fix or change how certificates are generated. Certs are expired --> need to re-generate new certs. 

I don't think we can realistically differentiate source vs remote in terms the source of issue, as we could be reaching out to ourselves, which is how one of the issues we hits manifests.

---

sounds good then! we should make sure that all of our upstream code doesn't barf on a new non transient error type

---

yeah we fail hard on a new non-transient error type (if thrown), so we will have to handle that :( I'm thinking we will cut a release adding this transient type, updating said dependencies, then releasing a new version of C*

---

although this may not be as bad as it seems, as we also do not handle `bootstrap_error`. https://github.com/palantir/cassandra/pull/319/files